airbyte-bootloader/src/main/java/io/airbyte/bootloader/BootloaderApp.java

```diff
@@ -72,32 +75,55 @@ public BootloaderApp(final Configs configs, final FeatureFlags featureFlags) {
    * @param configs
    * @param postLoadExecution
    */
-  public BootloaderApp(final Configs configs, final Runnable postLoadExecution, final FeatureFlags featureFlags) {
+  public BootloaderApp(final Configs configs,
+                       final Runnable postLoadExecution,
+                       final FeatureFlags featureFlags,
+                       final SecretMigrator secretMigrator) {
     this.configs = configs;
     this.postLoadExecution = postLoadExecution;
     this.featureFlags = featureFlags;
+    this.secretMigrator = secretMigrator;
   }
 
   public BootloaderApp() {
     configs = new EnvConfigs();
     featureFlags = new EnvVariableFeatureFlags();
-    postLoadExecution = () -> {
-      try {
-        final Database configDatabase =
-            new ConfigsDatabaseInstance(configs.getConfigDatabaseUser(), configs.getConfigDatabasePassword(), configs.getConfigDatabaseUrl())
-                .getAndInitialize();
-        final JsonSecretsProcessor jsonSecretsProcessor = JsonSecretsProcessor.builder()
-            .maskSecrets(!featureFlags.exposeSecretsInExport())
-            .copySecrets(true)
-            .build();
-        final ConfigPersistence configPersistence =
-            DatabaseConfigPersistence.createWithValidation(configDatabase, jsonSecretsProcessor);
-        configPersistence.loadData(YamlSeedConfigPersistence.getDefault());
-        LOGGER.info("Loaded seed data..");
-      } catch (final IOException e) {
-        e.printStackTrace();
+
+    try {
+      final Database configDatabase = new ConfigsDatabaseInstance(configs.getConfigDatabaseUser(), configs.getConfigDatabasePassword(),
+          configs.getConfigDatabaseUrl())
+              .getAndInitialize();
+      final JsonSecretsProcessor jsonSecretsProcessor = JsonSecretsProcessor.builder()
+          .maskSecrets(!featureFlags.exposeSecretsInExport())
+          .copySecrets(true)
+          .build();
+
+      final ConfigPersistence configPersistence = DatabaseConfigPersistence.createWithValidation(configDatabase, jsonSecretsProcessor);
+      final Optional<SecretPersistence> secretPersistence = SecretPersistence.getEphemeral(configs);
+      if (secretPersistence.isPresent()) {
```

Comment:
is there a reason we use this flow control pattern as opposed to just always running the `SecretMigrator` and having it be a no op if there is no persistence? Since we're trying to get all secrets handling down to one codepath, I'm trying to be a thoughtful about anywhere we are splitting flow control.

Reply:
Done,